articles/151_roslaunch_xml.md

```diff
@@ -0,0 +1,744 @@
+            the predicate evaluates to a truthy value.
+          </xs:documentation>
+        </xs:annotation>
+      </xs:attribute>
+  </xs:element>
+</xs:schema> 
+```
+
+### Tags Semantics
+
+#### All Tags
+
+Every tag's execution can be conditioned on a boolean predicate via `if` or `unless` attributes.
+
+#### <launch> Tag
+
+Root tag of any launch file.
+
+#### <include> Tag
+
+##### Description
+
+The <include> tag allows for bringing a launch file description into another, enabling re-use of hierarchical launch layouts.
+The included launch file description has its own scope for launch configurations.
+The included launch file description is not necessarily written in this format nor a declarative one (i.e. a programmatic description).
+To avoid ROS name clashes, included nodes can be namespaced via the `ns` attribute.
+
+##### Examples
+
+```xml
+<include file="/opt/my_launch_file.py" ns="/my_ns"/>
+<include file="/opt/my_other_launch_file.xml">
+  <arg name="some_argument" value="dummy_value"/>
+</include>
+```
+
+#### <group> Tag
+
+##### Description
+
+The <group> tag allows for launch actions' grouping as well as optional launch file configuration scoping.
+
+##### Examples
+
+```xml
+<group ns="/dummy_group" scoped="true">
+  <node package="a_ros_package" name="dummy0" executable="dummy_node"/>
+  <node package="a_ros_package" name="dummy1" executable="dummy_node"/>
+</group>
+```
+
+#### <arg> Tag
+
+##### Description
+
+The <arg> tag allows for launch file configuration via the command-line or when including it via an <include> tag.
+Arguments are limited to the scope of their definition and thus, have to be explictly passed to included files if any.
+
+##### Examples
+
+```xml
+<arg name="publish_frequency" default="10"/>
+<arg name="output_path" doc="Output path for some processing pipeline"/>
+```
+
+#### <node> Tag
+
+##### Description
+
+The <node> tag allows for executing a ROS node in the form of local OS process.
+
+##### Examples
+
+```xml
+<node package="ros_demos" executable="publisher">
+  <param name="publish_frequency">10</param>
+  <remap from="generic_topic_name" to="my_topic"/>
+</node>
+```
+
+#### <param> Tag
+
+##### Description
+
+The <param> tag allows for setting a ROS parameter of a ROS node.
+
+##### Examples
+
+```xml
+<param name="some_numeric_param" value="100.2"/>
+<param name="some_list_param" value="Some phrase,100.0,true" sep=","/>
+```
+
+#### <params> Tag
+
+##### Description
+
+The <params> tag allows to either bring ROS parameters from a YAML parameters file or to nest a <param> definitions under an appropriate name (i.e. to make a map of them).
+
+##### Examples
+
+```xml
+<params from="path/to/param/file.yml"/>
+<params ns="some_param_group">
+  <param name="some_integer_param" value="10"/>
+</params>
+```
+
+#### <remap> Tag
+
+##### Description
+
+The <remap> tag allows for ROS name remapping of a <node> instance.
+
+##### Examples
+
+```xml
+<remap from="chatter" to="my_chatter"/>
```

Comment:
Again it would be useful to show that this must be in a node tag, even if state otherwise.

Reply:
Same as above.